lib/services/plugins-service.ts

```diff
@@ -0,0 +1,190 @@
+						this.$fs.writeFile(platformData.configurationFilePath, resultXml).wait();	
+					}
+					
+					if(this.$fs.exists(pluginPlatformsFolderPath).wait()) {
+						shelljs.rm("-rf", pluginPlatformsFolderPath);			
+					}						
+					
+					// TODO: Add libraries
+					
+					// Show message
+					this.$logger.out(`Successfully prepared plugin ${pluginData.name} for ${platform}.`);
+				}
+			};
+			
+			this.executeForAllInstalledPlatforms(action);
+		}).future<void>()();
+	}
+	
+	public getAllInstalledPlugins(): IFuture<IPluginData[]> {
+		return (() => {
+			let nodeModules = this.$fs.readDirectory(this.nodeModulesPath).wait();
+			let plugins: IPluginData[] = [];
+			_.each(nodeModules, nodeModuleName => {
+				var nodeModuleData = this.getNodeModuleData(nodeModuleName);
+				if(nodeModuleData.isPlugin) {
+					plugins.push(this.convertToPluginData(nodeModuleData));
+				}
+			});
+			
+			return plugins;
+		}).future<IPluginData[]>()();
+	}
+	
+	private executeNpmCommand(npmCommandName: string, npmCommandArguments: string): IFuture<string> {
+		return (() => {
+			let command = this.composeNpmCommand(npmCommandName, npmCommandArguments);
+			let result = this.$childProcess.exec(command, { cwd: this.$projectData.projectDir }).wait();
+			return this.parseNpmCommandResult(result);
+		}).future<string>()();
+	}
+	
+	private composeNpmCommand(npmCommandName: string, npmCommandArguments: string): string {
+		let command = `npm ${npmCommandName} ${npmCommandArguments} --save `;
+		if(this.$options.production) {
+			command += " --production ";
+		}
+		
+		return command;
+	}
+	
+	private parseNpmCommandResult(npmCommandResult: string): string { // The npmCommandResult is in the following format: [<name>@<version node_modules/<name>]
+		return npmCommandResult.split("@")[0]; // returns plugin name
+	}
+	
+	private executeForAllInstalledPlatforms(action: (pluginDestinationPath: string, platform: string, platformData: IPlatformData) => void): void {
+		let availablePlatforms = _.keys(this.$platformsData.availablePlatforms);
+		_.each(availablePlatforms, platform => {
+			let isPlatformInstalled = this.$fs.exists(path.join(this.$projectData.platformsDir, platform.toLowerCase())).wait();
+			if(isPlatformInstalled) {
+				let platformData = this.$platformsData.getPlatformData(platform.toLowerCase());
+				let pluginDestinationPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME, "tns_modules");
+				action(pluginDestinationPath, platform.toLowerCase(), platformData);
+			}
+		});
+	}
+	
+	private getNodeModuleData(moduleName: string): INodeModuleData {
+		let fullNodeModulePath = path.join(this.nodeModulesPath, moduleName);
+		let packageJsonFilePath = path.join(fullNodeModulePath, "package.json");
+		let data = require(packageJsonFilePath);
+		let result = {
+			name: data.name,
+			version: data.version,
+			isPlugin: data.nativescript,		
```

Comment:
isPlugin suggests a boolean value, but here is the entire object. I suggest you either give this property a boolean value, or rename it to something like "pluginInfo".